File: Data Structures & Algorithms/depth-of-binary-tree/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def dfs(self,root):
        if root is None:return 0
        left=1+self.dfs(root.left)
        right=1+self.dfs(root.right)
        return max(left,right)

    def maxDepth(self, root: Optional[TreeNode]) -> int:
        if root is None:return 0;
        return self.dfs(root)
        

File: Data Structures & Algorithms/longest-increasing-subsequence/submission-2.cpp
class Solution {
public:
int dfs(int n,int i,int j,vector<int> &nums,vector<vector<int>> &dp){
    // base case:
    if(i==n) return 0;
    if(dp[i][j+1]!=-1) return dp[i][j+1];
    int LIS=dfs(n,i+1,j,nums,dp);    // Not include
    if(j==-1 || nums[i] > nums[j]) {
        LIS=max(LIS,1+dfs(n,i+1,i,nums,dp))  ;        //include
    }
    return dp[i][j+1]=LIS;
   
}
    int lengthOfLIS(vector<int>& nums) {
        int n=nums.size();
        vector<vector<int>> dp(n,vector<int>(n,-1));
        return dfs(n,0,-1,nums,dp);

        
    }
};
